### PR TITLE
[SassTask] [core] - Fixed Filesystem::listContents for numeric filenames.

### PR DESCRIPTION
**Pull request by malhusseiny**
Fixed an issue that affects all filesystem manipulation tasks dealing with files and directories with all-numeric names and often causes build failures and/or infinite directory traversal loops. Added a simple test for the fix.

Fixed the FailOnError test bug.

### CODE REVIEW

File: src/Phing/Io/FileSystem.php [hunk 1]
```diff
@@ -869,14 +869,17 @@ public function compareMTimes($file1, $file2)
...
+        $filenames = array_keys(
             iterator_to_array(
                 new FilesystemIterator(
                     $f->getAbsolutePath(),
                     FilesystemIterator::KEY_AS_FILENAME
                 )
             )
         );
+
+        // make sure numeric filenames (e.g. 11) are properly treated as strings
```
Comment (siad007) [inline src/Phing/Io/FileSystem.php:881]: Please remove - this is covered in your test.

File: src/Phing/Io/FileSystem.php [hunk 2]
```diff
@@ -869,14 +869,17 @@ public function compareMTimes($file1, $file2)
      */
     public function listContents(File $f)
     {
-        return array_keys(
+        $filenames = array_keys(
```
Comment (siad007) [inline src/Phing/Io/FileSystem.php:872]: No need for this temporary variable - just do it all in one :)

**Review by siad007 — CHANGES_REQUESTED** [review]
Thanks for your contribution @malhusseiny - small change requests from my side :)

File: src/Phing/Io/FileSystem.php (hunk 2 above)
Comment (malhusseiny) [reply src/Phing/Io/FileSystem.php]: Done :)

**Review by siad007 — APPROVED** [review]

**Comment by siad007** [comment]
@malhusseiny LGTM - thx

**Pull request merged**